Sources/XMTP/Conversations.swift

```diff
@@ -79,16 +81,20 @@ public class Conversations {
 
                     do {
                         for try await envelope in client.subscribe(topics: topics) {
-                            if let conversation = conversationsByTopic[envelope.contentTopic] {
+                            if let conversation = await conversationsByTopic[envelope.contentTopic] {
                                 let decoded = try conversation.decode(envelope)
                                 continuation.yield(decoded)
                             } else if envelope.contentTopic.hasPrefix("/xmtp/0/invite-") {
                                 let conversation = try fromInvite(envelope: envelope)
-                                conversationsByTopic[conversation.topic] = conversation
+                                async {
```

Comment:
I think these should be `await`s, not async tasks. If you make a task here, the loop can break before the dictionary value is written, so it will miss the subscription.